xilem/examples/calc.rs

```diff
@@ -0,0 +1,326 @@
+                .flex(1.0),
+            sized_box(button("DEL", |data: &mut AppData| data.on_delete()))
+                .expand()
+                .flex(1.0),
+            operator_button(MathOperator::Divide).flex(1.0),
+        ))
+        .direction(Axis::Horizontal)
+        .cross_axis_alignment(CrossAxisAlignment::Center)
+        .main_axis_alignment(MainAxisAlignment::SpaceEvenly)
+        .gap(GRID_GAP)
+        .flex(1.0),
+        // 7 8 9 X
+        flex((
+            digit_button("7").flex(1.0),
+            digit_button("8").flex(1.0),
+            digit_button("9").flex(1.0),
+            operator_button(MathOperator::Multiply).flex(1.0),
+        ))
+        .direction(Axis::Horizontal)
+        .cross_axis_alignment(CrossAxisAlignment::Fill)
+        .main_axis_alignment(MainAxisAlignment::SpaceEvenly)
+        .gap(GRID_GAP)
+        .flex(1.0),
+        // 4 5 6 -
+        flex((
+            digit_button("4").flex(1.0),
+            digit_button("5").flex(1.0),
+            digit_button("6").flex(1.0),
+            operator_button(MathOperator::Subtract).flex(1.0),
+        ))
+        .direction(Axis::Horizontal)
+        .cross_axis_alignment(CrossAxisAlignment::Fill)
+        .main_axis_alignment(MainAxisAlignment::SpaceEvenly)
+        .gap(GRID_GAP)
+        .flex(1.0),
+        // 1 2 3 +
+        flex((
+            digit_button("1").flex(1.0),
+            digit_button("2").flex(1.0),
+            digit_button("3").flex(1.0),
+            operator_button(MathOperator::Add).flex(1.0),
+        ))
+        .direction(Axis::Horizontal)
+        .cross_axis_alignment(CrossAxisAlignment::Fill)
+        .main_axis_alignment(MainAxisAlignment::SpaceEvenly)
+        .gap(GRID_GAP)
+        .flex(1.0),
+        // bottom row
+        flex((
+            sized_box(button("±", |data: &mut AppData| data.negate()))
+                .expand()
+                .flex(1.0),
+            digit_button("0").flex(1.0),
+            digit_button(".").flex(1.0),
+            sized_box(button("=", |data: &mut AppData| data.on_equals()))
+                .expand()
+                .flex(1.0),
+        ))
+        .direction(Axis::Horizontal)
+        .cross_axis_alignment(CrossAxisAlignment::Fill)
+        .main_axis_alignment(MainAxisAlignment::SpaceEvenly)
+        .gap(GRID_GAP)
+        .flex(1.0),
+    ))
+    .gap(GRID_GAP)
+    .cross_axis_alignment(CrossAxisAlignment::Fill)
+    .main_axis_alignment(MainAxisAlignment::End)
+    .must_fill_major_axis(true)
+}
+fn operator_button(math_operator: MathOperator) -> impl WidgetView<AppData> {
+    sized_box(button(math_operator.as_str(), move |data: &mut AppData| {
+        data.on_entered_operator(math_operator);
+    }))
+    .expand()
+}
+
+fn digit_button(digit: &'static str) -> impl WidgetView<AppData> {
+    sized_box(button(digit, |data: &mut AppData| {
+        data.on_entered_digit(digit);
+    }))
+    .expand()
+}
+
+fn play_bell_sound() {
+    println!("\x07");
```

Comment:
Yeah I think that should probably done differently, if at all, so I would just remove it for now, maybe add a visual indicator or something, but as mentioned that is not really easy to support cleanly currently.

But it raises an interesting question regarding playing system sounds (also interesting cross-platform I guess).

Reply:
The point I'm making is that this doesn't actually make a sound for me, so the UX is that there is no feedback at all. I don't have a great suggestion - probably a `prose` with red text in a line above (below?) the main calculation line.

It's probably better to demonstrate this kind of "one-time" state within the app anyway.

Yes, ideally it would make a sound, but we need that to be a cross-platform way.

Reply:
I removed it, and put comments where the calls were originally for future work.